IntegrationWithPharmacies/MedicineInformationApi/Service/MedicineInformationService.cs

```diff
@@ -0,0 +1,109 @@
+﻿using MedicineInformationApi.Adapter;
+using MedicineInformationApi.DbContextModel;
+using MedicineInformationApi.Dto;
+using MedicineInformationApi.Model;
+using MedicineInformationApi.Repository;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MedicineInformationApi.Service
+{
+    public class MedicineInformationService : IMedicineInformationService
+    {
+        public MedicineInformationRepository MedicineWithQuantityRepository { get; }
+        public IMedicineInformationRepository IMedicineWithQuantityRepository { get; }
+        public MedicineInformationService() { }
+
+        public MedicineInformationService(MyDbContext context)
+        {
+            MedicineWithQuantityRepository = new MedicineInformationRepository(context);
+        }
+        public MedicineInformationService(IMedicineInformationRepository mdicineWithQuantityRepository)
+        {
+            IMedicineWithQuantityRepository = mdicineWithQuantityRepository;
+        }
+
+
+        public List<MedicineInformation> GetAll()
+        {
+            return MedicineWithQuantityRepository.GetAll();
+        }
+
+        public MedicineInformation Create(MedicineInformationDto medicineWithQuantityDto)
+        {
+            return MedicineWithQuantityRepository.Create(MedicineInformationAdapter.MedicineWithQuantityDtoToMedicineWithQuantity(medicineWithQuantityDto));
+        }
+
+        public string GetMedicineDescriptionFromStub(string medicine)
+        {
+            MedicineInformation medicineWithQuantity = GetAllForStub().SingleOrDefault(medicine1 => medicine1.MedicineDescription.Name.Equals(medicine));
+            return (medicineWithQuantity != null ? medicineWithQuantity.MedicineDescription.Description : "");
+        }
+
+        public List<MedicineInformation> GetAllForStub()
+        {
+            return IMedicineWithQuantityRepository.GetAll();
+        }
+
+        public List<MedicineName>  GetAllMedicinesFromDatabase()
+        {
+            List<MedicineInformation> medicines = MedicineWithQuantityRepository.GetAll();
+            return getMedicineNames(medicines);
```

Comment:
Write this in form:
"return getMedicineNames(MedicineWithQuantityRepository.GetAll());"